LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A01_t01.dart

```diff
@@ -0,0 +1,125 @@
+    return new ArgumentsBindingClass.named(t1, t2: t2);
+  }
+  factory ArgumentsBindingClass.fPositional((num, num, String?) t1, [(num, num, String?) t2 = t1Default]) {
+    return new ArgumentsBindingClass.positional(t1, t2);
+  }
+
+  static namedArgumentsStaticMethod((num, num, String?) t1, {(num, num, String?) t2 = t1Default}) {}
+  static positionalArgumentsStaticMethod((num, num, String?) t1, [(num, num, String?) t2 = t1Default]) {}
+
+  namedArgumentsMethod((num, num, String?) t1, {(num, num, String?) t2 = t1Default}) {}
+  positionalArgumentsMethod((num, num, String?) t1, [(num, num, String?) t2 = t1Default]) {}
+
+  set testSetter((num, num, String?) val) {}
+}
+
+class ArgumentsBindingGen<X>  {
+  ArgumentsBindingGen(X t1) {}
+
+  ArgumentsBindingGen.named(X t1, {required X t2}) {}
+
+  factory ArgumentsBindingGen.fNamed(X t1, {required X t2}) {
+    return new ArgumentsBindingGen.named(t1, t2: t2);
+  }
+
+  namedArgumentsMethod(X t1, {required X t2}) {}
+
+  set testSetter(X val) {}
+}
+
+main() {
+  // test functions
+  namedArgumentsFunc1(forgetType(t0Instance), t2: forgetType(t0Instance));
+  positionalArgumentsFunc1(forgetType(t0Instance), forgetType(t0Instance));
+
+  // test class constructors
+  ArgumentsBindingClass instance1 =
+      new ArgumentsBindingClass(forgetType(t0Instance));
+  instance1 = new ArgumentsBindingClass.fNamed(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  instance1 = new ArgumentsBindingClass.named(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  instance1 = new ArgumentsBindingClass.positional(forgetType(t0Instance),
+      forgetType(t0Instance));
+
+  // tests methods and setters
+  instance1.namedArgumentsMethod(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  instance1.positionalArgumentsMethod(forgetType(t0Instance),
+      forgetType(t0Instance));
+  instance1.testSetter = forgetType(t0Instance);
+
+  // test static methods
+  ArgumentsBindingClass.namedArgumentsStaticMethod(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  ArgumentsBindingClass.positionalArgumentsStaticMethod(
+      forgetType(t0Instance), forgetType(t0Instance));
+
+  // Test type parameters
+
+  //# <-- NotGenericFunctionType
+  // test generic functions
+  namedArgumentsFunc2<(num, num, String?)>(forgetType(t0Instance), t2: forgetType(t0Instance));
+
+  // test generic class constructors
+  ArgumentsBindingGen<(num, num, String?)> instance2 =
+      new ArgumentsBindingGen<(num, num, String?)>(forgetType(t0Instance));
+  instance2 = new ArgumentsBindingGen<(num, num, String?)>.fNamed(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  instance2 = new ArgumentsBindingGen<(num, num, String?)>.named(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+
+  // test generic class methods and setters
+  instance2.namedArgumentsMethod(forgetType(t0Instance),
+      t2: forgetType(t0Instance));
+  instance2.testSetter = forgetType(t0Instance);
+  //# -->
+}
```

Comment:
`t1Instance` is never used? Why was it declared, then?

Reply:
I understand that this kind of thing may more difficult to avoid when the code is generated. But the code generator might still be able to keep track of whether or not `t1Instance` has been used, and include the declaration of `t1Instance` (and other things) if and only if they've been used.

Reply:
This is a generated test. It is generated from `records_A01.dart`  and  `arguments_binding_x01.dart`. In the first file we define types `T0` and `T1` and instance variables
```dart
T0 t0Instance = ...;
T1 t1Instance = ...;
```
File `arguments_binding_x01.dart` checks that `T0` is subtype of `T1` in an arguments binding area. In other words it chacks that `t0Instance` can be used everywhere where we expect argument of type `T1`. Test generator combines these two files in one test. In this particular case it turned out that `tiInstance` is not used in this test. But it is used in other tests. In theory it's possible to enhance tests generartor to catch these cases and remove declaration of `t1Instance`  in such cases but it is a separate change

Reply:
OK!